linux/mcompile/keymap/keymap.cpp

```diff
@@ -0,0 +1,426 @@
+  else if ( in == "key<AD09>")    out = 32;
+  else if ( in == "key<AD10>")    out = 33;
+  else if ( in == "key<AD11>")    out = 34;
+  else if ( in == "key<AD12>")    out = 35;
+
+  else if ( in == "key<AC01>")    out = 38;
+  else if ( in == "key<AC02>")    out = 39;
+  else if ( in == "key<AC03>")    out = 40;
+  else if ( in == "key<AC04>")    out = 41;
+  else if ( in == "key<AC05>")    out = 42;
+  else if ( in == "key<AC06>")    out = 43;
+  else if ( in == "key<AC07>")    out = 44;
+  else if ( in == "key<AC08>")    out = 45;
+  else if ( in == "key<AC09>")    out = 46;
+  else if ( in == "key<AC10>")    out = 47;
+  else if ( in == "key<AC11>")    out = 48;
+  else if ( in == "key<AC12>")    out = 49;
+
+  else if ( in == "key<AB01>")    out = 52;
+  else if ( in == "key<AB02>")    out = 53;
+  else if ( in == "key<AB03>")    out = 54;
+  else if ( in == "key<AB04>")    out = 55;
+  else if ( in == "key<AB05>")    out = 56;
+  else if ( in == "key<AB06>")    out = 57;
+  else if ( in == "key<AB07>")    out = 58;
+  else if ( in == "key<AB08>")    out = 59;
+  else if ( in == "key<AB09>")    out = 60;
+  else if ( in == "key<AB10>")    out = 61;
+  else if ( in == "key<BKSL>")    out = 62;   //correct ???
+  else if ( in == "key<LSGT>")    out = 51;   //correct ???
+  return out;
+}
+
+void append_other_ToVector(v_str_3D &All_Vector,GdkKeymap * keymap) {
+
+  // create a 2D vector all fill0ed with "--" and push to 3D-Vector
+  v_str_2D Other_Vector2D = create_empty_2D(All_Vector[0].size(),All_Vector[0][0].size());
+  All_Vector.push_back(Other_Vector2D);
+
+  printf("+++++++ dimensions of Vector after append_other_ToVector\t %li..%li..%li\n", All_Vector.size(), All_Vector[0].size(),All_Vector[0][0].size());  
+
+  for(int i =1; i< (int) All_Vector[1].size()-1;i++)
+  {
+    // get key name US stored in [0][i][0] and copy to name in other-block[1][i][0]
+    All_Vector[1][i][0] = All_Vector[0][i][0];
+
+    // TODO see that nr of shift states are not out of range !!
+    // write this value to 3D- Vector
+    All_Vector[1][i][0+1] = GetKeyvalsFromKeymap(keymap,stoi(All_Vector[1][i][0]),0);   //shift state: unshifted:0
+    All_Vector[1][i][1+1] = GetKeyvalsFromKeymap(keymap,stoi(All_Vector[1][i][0]),1);   //shift state: shifted:1
+    All_Vector[1][i][2+1] = GetKeyvalsFromKeymap(keymap,stoi(All_Vector[1][i][0]),2);   //shift state: ?
+    All_Vector[1][i][3+1] = GetKeyvalsFromKeymap(keymap,stoi(All_Vector[1][i][0]),3);   //shift state: ?
+
+    //printf("Keycodes US->Other:   %d(US): %s %s ---- (other):%s,  %s,  %s  \n",stoi(All_Vector[1][i][0]),All_Vector[0][i][1].c_str(),All_Vector[0][i][2].c_str(),All_Vector[1][i][1].c_str(),All_Vector[1][i][2].c_str(),All_Vector[1][i][3].c_str());  
+  }
+}
+
+v_str_2D create_empty_2D( int dim_rows,int dim_shifts)
+{
+  std::string empty = "--";
+  v_str_1D shifts;
+  v_str_2D all;
+
+  for ( int i=0; i< dim_rows;i++) {
+    for ( int i=0; i< dim_shifts;i++) {
```

Comment:
It might be less confusing (and easier to debug) if you'd use a different variable name for the nested loop, e.g. `j`.